CRM/Financial/BAO/Order.php

```diff
@@ -1122,4 +1123,20 @@ protected function fillDefaultContributionLine(array &$lineItem): void {
     $lineItem = array_merge($defaults, $lineItem);
   }
 
+  /**
+   * Update line item to have amount & tax amount based off tax rate & inclusive amount.
+   *
+   * @param float $taxRate
+   * @param array $lineItem
+   * @param $inclusiveAmount
+   *
+   * @return array
+   */
+  protected function alterLineItemForTax($taxRate, array $lineItem, $inclusiveAmount): array {
+    // Total is tax inclusive.
+    $lineItem['tax_amount'] = ($taxRate / 100) * $inclusiveAmount / (1 + ($taxRate / 100));
+    $lineItem['line_total'] = $lineItem['unit_price'] = $inclusiveAmount - $lineItem['tax_amount'];
+    return $lineItem;
+  }
```

Comment:
Will wait for the Jenkins to run first then pull this in.

Reply:
Ah - those are those test fails that I thought would happen - this gets me back to 'this api has always behaved differently depending on ... something'

@KarinG a couple of notes to your usage
1) we need to make sure the rc doesn't break your usage & find way to do that
2) against master we would ideally get your code to call the order api correctly (ie if there is more than one line item then pass them in OR (less good) just switch to doing exactly the same code but call Contribution.create with contribution_status_id = Pending which I think is functionally equivalent the way you are using the code (given you then delete & recreate the line items). I'd like to think that would be an interim thing & I'm totally happy to work with you on getting to 'full line item'


```
Stacktrace
api_v3_ContributionTest::testCheckTaxAmount with data set #0 ('.')
Failed asserting that '4.76' matches expected 5.

/home/jenkins/bknix-dfl/build/core-21187-75kjs/web/sites/all/modules/civicrm/tests/phpunit/api/v3/ContributionTest.php:2284
/home/jenkins/bknix-dfl/build/core-21187-75kjs/web/sites/all/modules/civicrm/tests/phpunit/CiviTest/CiviUnitTestCase.php:258
/home/jenkins/bknix-dfl/extern/phpunit8/phpunit8.phar:671

```

Reply:
@KarinG I think https://github.com/civicrm/civicrm-core/pull/21187/commits/0f4482bc116fb83e1985352e6c05ec28e03bd989 will allow your override tax amount to keep working until we have time to address it - let's see how tests go. 

Also your work clarifying the contribution_recur was useful in knowing how to deal with the test fail :-)